### PR TITLE
Report exporter role

### PR DESCRIPTION
**Pull request by KrekoG**
## Features
 - Restrict report downloads for Download Reports permission

[Delivers: [#140855703](https://www.pivotaltracker.com/story/show/140855703)]

### CODE REVIEW

**Comment by KrekoG** [comment]
@EvanBarbour3 Ready for your review!

File: app/Http/Middleware/Download.php [hunk 1]
```diff
@@ -39,7 +40,7 @@ public function handle($request, Closure $next)
         $source = $request->get('source', 'api_data');
         $filename = $request->get('filename', 'export_' . date('Y-m-d_Hi'));
 
-        if ($request->get('download') && array_key_exists($source, $response->original->getData())) {
```
Comment (EvanBarbour3) [inline app/Http/Middleware/Download.php]: What if Auth::user() returns null?

Comment (KrekoG) [inline app/Http/Middleware/Download.php]: In theory the user should only be able to download files when logged in, but good spot, a safety check should be in place

**Comment by KrekoG** [comment]
@EvanBarbour3 Ready for your review.

**Pull request merged**